core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/util/TransformStreamSupplier.java

```diff
@@ -54,10 +58,24 @@ public void close() throws IOException {
 
     @Override
     public Stream<Element> get() {
+        final Map<String, ElementTransformer> edgesTransform = transform.getEdges();
+        final Map<String, ElementTransformer> entitiesTransform = transform.getEntities();
 
         final Function<Element, Element> toTransformedElement = e -> {
-            final ElementTransformer elementTransformer = e instanceof Edge ? transform.getEdges().get(e.getGroup()) : transform.getEntities().get(e.getGroup());
-            return elementTransformer.apply(e);
+            final String group = e.getGroup();
+            final Element rtn;
+
+            if (e instanceof Edge && nonNull(edgesTransform) && nonNull(group)) {
+                rtn = edgesTransform.get(group).apply(e);
+            } else if (e instanceof Entity && nonNull(entitiesTransform) && nonNull(group)) {
+                rtn = entitiesTransform.get(group).apply(e);
+            } else if (e instanceof Edge || e instanceof Entity) {
+                //either the group or transformer was null for the Edge/Entity.
+                rtn = e;
+            } else {
+                throw new UnsupportedOperationException("Transform is only compatible with Edge or Entity elements, found: " + e.getClass());
```

Comment:
This is probably the wrong exception type. UnsupportedOperationExceptions tend to only be thrown when the user tries to execute an operation which is not supported, as opposed to when the input for an operation is the wrong type. An IllegalArgumentException may be more appropriate.